packages/shared/src/effect.ts

```diff
@@ -24,72 +24,72 @@ export class FetchContext extends Context.Tag("uploadthing/FetchContext")<
   FetchContextService
 >() {}
 
-interface ResponseWithBoundRequest extends ResponseEsque {
-  request: Request;
+interface ResponseWithURL extends ResponseEsque {
+  requestUrl: string;
 }
 
 // Temporary Effect wrappers below.
 // TODO should be refactored with much love
 // TODO handle error properly
 export const fetchEff = (
-  input: RequestInfo | URL,
+  input: string | URL,
   init?: RequestInit,
-): Effect.Effect<ResponseWithBoundRequest, FetchError, FetchContext> =>
+): Effect.Effect<ResponseWithURL, FetchError, FetchContext> =>
   Effect.flatMap(FetchContext, ({ fetch, baseHeaders }) => {
-    const req = new Request(input, {
-      ...init,
+    const reqInfo = {
+      url: input.toString(),
+      method: init?.method,
+      body: init?.body,
       headers: {
         ...filterObjectValues(baseHeaders, (v): v is string => v != null),
         ...init?.headers,
       },
-    });
+    };
     return Effect.tryPromise({
-      try: () => fetch(req.clone()),
-      catch: (error) => new FetchError({ error, input: req.clone() }),
+      try: () => fetch(input, { ...init, headers: reqInfo.headers }),
+      catch: (error) => new FetchError({ error, input: reqInfo }),
     }).pipe(
-      Effect.map((res) => Object.assign(res.clone(), { request: req.clone() })),
-      Effect.withSpan("fetch", {
-        attributes: { input: JSON.stringify(input) },
-      }),
+      Effect.map((res) => Object.assign(res, { requestUrl: reqInfo.url })),
+      Effect.withSpan("fetch", { attributes: { reqInfo } }),
     );
   });
 
-export const json = (
-  res: ResponseWithBoundRequest,
-): Effect.Effect<unknown, FetchError | BadRequestError<Json>, never> =>
+export const parseJson = (
+  res: ResponseWithURL,
+): Effect.Effect<unknown, InvalidJsonError | BadRequestError, never> =>
```

Comment:
the json is being parsed, no doubt about it. But it needs to be parsed again later to turn it from an `unknown` to some more significant shape, which is why I chose `json`.  Since it just turns a bit array into a json entity-- or die trying

Reply:
valid point